src/main/java/bio/terra/app/model/GoogleRegion.java

```diff
@@ -69,4 +80,27 @@ public static GoogleRegion fromValue(String text) {
     public static GoogleRegion fromValueWithDefault(String text) {
         return Optional.ofNullable(GoogleRegion.fromValue(text)).orElse(GoogleRegion.DEFAULT_GOOGLE_REGION);
     }
+
+    public static boolean matchingRegionWithFallbacks(List<StorageResource> storage, GoogleRegion region) {
+        GoogleRegion bqRegion = region;
+        GoogleRegion firestoreRegion = region.getRegionOrFallbackFirestoreRegion();
+        GoogleRegion bucketRegion = region.getRegionOrFallbackBucketRegion();
+        boolean matching = true;
+        for (StorageResource resource : storage) {
+            if (resource.getCloudResource() == GoogleCloudResource.BIGQUERY) {
+                if (resource.getRegion() != bqRegion) {
+                    matching = false;
+                }
+            } else if (resource.getCloudResource() == GoogleCloudResource.FIRESTORE) {
+                if (resource.getRegion() != firestoreRegion) {
+                    matching = false;
+                }
+            } else if (resource.getCloudResource() == GoogleCloudResource.BUCKET) {
+                if (resource.getRegion() != bucketRegion) {
+                    matching = false;
+                }
+            }
+        }
+        return matching;
```

Comment:
Two things: using `switch` is a better way to write `if .. else if ..`` when you're comparing the same object over and over again, as it makes it easier to see that that's what going on. Also, once `matching` is `false` its value never changes back to `true`, so there's no need to continue the loop at that point.
```suggestion
        for (StorageResource resource : storage) {
            switch (resource.getCloudResource()) {
            case BIGQUERY:
                if (resource.getRegion() != bqRegion) {
                    return false;
                }
                break;
            case FIRESTORE:
                if (resource.getRegion() != firestoreRegion) {
                    return false;
                }
                break;
            case BUCKET:
                if (resource.getRegion() != bucketRegion) {
                   return false;
                }
            }
        }
        return true;
```

Reply:
Won't the `break` statements here refer to the `switch` statement instead of the `for` loop?

Reply:
Oh, returning to break. Duh.